src/convertToObject.js

```diff
@@ -6,7 +6,34 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesList = sourceString.replaceAll('\n', ' ').split(';');
+  const filtredStylesList = stylesList.filter((val) => val.trim().length > 0);
+
+  const segregatedList = [];
+  const stylesObject = {};
+
+  function segregate(el) {
+    const splited = el.split(':');
+
+    for (let i = 0; i < splited.length; i++) {
+      if (splited[i].length > 0) {
+        segregatedList.push(splited[i].trim());
+      }
+    }
```

Comment:
The use of a `for` loop here does not meet the task requirement of using iteration methods like `map` or `filter`. Consider refactoring this logic to use `map` or `filter` instead.